tools/cache.py

```diff
@@ -45,10 +45,14 @@ def try_remove_ending(thestring, ending):
     if use_subdir:
       if not shared.Settings.WASM_BACKEND:
         dirname = os.path.join(dirname, 'asmjs')
-      elif shared.Settings.WASM_OBJECT_FILES:
-        dirname = os.path.join(dirname, 'wasm_o')
       else:
-        dirname = os.path.join(dirname, 'wasm_bc')
+        dirname = os.path.join(dirname, 'wasm')
+        if shared.Settings.WASM_OBJECT_FILES:
+          dirname = os.path.join(dirname, 'obj')
+        else:
+          dirname = os.path.join(dirname, 'bc')
+        if shared.Settings.RELOCATABLE:
+          dirname = os.path.join(dirname, 'pic')
```

Comment:
do we need `pic-obj`, `pic-bc`?

Reply:
This code already takes care of that by just adding another level of nesting.  However I decided to flatten it out.